### PR TITLE
Add FLUX image-to-image and inpainting

### PR DESCRIPTION
**Pull request by RyanJDick**
## Summary

This PR adds support for Image-to-Image and inpainting workflows with the FLUX model.

Full changelog:
- Split out `FLUX VAE Encode` and `FLUX VAE Decode` nodes
- Renamed `FLUX Text-to-Image` node to `FLUX Denoise` (since it now supports image-to-image too). This is a workflow-breaking change.
- Added support for FLUX image-to-image via the `Latents` param on the FLUX denoising node.
- Added support for FLUX masked inpainting via the `Denoise Mask` param on the FLUX denoising node.
- Added "Denoise Start" and "Denoise End" params to the "FLUX Denoise" node.
- Updated the "FLUX Text to Image" default workflow.
- Added a "FLUX Image to Image" default workflow.

### Example

FLUX inpainting workflow
image

Input image
image

Mask
image

Output image
image

### Callouts for reviewers:
- I renamed FLUXTextToImageInvocation -> FLUXDenoisingInvocation. This is, of course, a breaking change. It feels like the right move and now is the right time to do it. Any objection?
- I added new `FLUX VAE Encode` and `FLUX VAE Decode` nodes. Alternatively, I could have tried to match these names to the corresponding SD nodes (e.g. `FLUX Image to Latents`, `FLUX Latents to Image`). Personally, I prefer the current names, but want to hear other opinions.

### Usage notes:
- With the default dev timestep scheduler, the image structure is largely determined in the first ~3 steps. A consequence of this is that the denoise_start parameter provides limited 'granularity' of control. This will likely be improved in the future as we add more scheduler options. In the meantime, you will likely want to use small values for `denoise_start` (e.g. 0.03) to start denoising on step ~1-4 out of ~30.
- Currently, there is no 'noise' parameter on the `FLUX Denoise` node, so the `denoise_end` parameter has limited utility. This will be added in the future.

## QA Instructions

Test the following workflows:
- [x] Vanilla FLUX text-to-image behaviour is unchanged
- [x] Image-to-image with FLUX dev, no mask
- [x] Image-to-image with FLUX dev, with mask
- [x] Image-to-image with FLUX schnell, no mask (smoke test, not expected to work well)

## Merge Plan

No special instructions.

## Checklist

- [x] _The PR has a short but descriptive title, suitable for a changelog_
- [x] _Tests added / updated (if applicable)_
- [x] _Documentation added / updated (if applicable)_

### CODE REVIEW

**Review by hipsterusername — COMMENTED** [review]
The only consideration here is inconsistency between name of VAE between flux and SD nodes. I think the simpler name is probably the best top level name, but it should be discoverable under encode

**Review by hipsterusername — APPROVED** [review]
Will approve in case the desire is to address this later

**Review by brandonrising — APPROVED** [review]
Other than that comment lgtm! It's somewhat tangential to the work here so it could be updated in a separate PR

**Comment by RyanJDick** [comment]
> The only consideration here is inconsistency between name of VAE between flux and SD nodes. I think the simpler name is probably the best top level name, but it should be discoverable under encode

Renamed 'FLUX VAE Encode'/'FLUX VAE Decode' to 'FLUX Image to Latents'/'FLUX Latents to Image' in [b43da46](https://github.com/invoke-ai/InvokeAI/pull/6798/commits/b43da46b829ccb46e13ff67084da6170eb1b2469)

File: invokeai/app/services/workflow_records/default_workflows/FLUX Image to Image.json [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+      "nodeId": "f8d9d7c8-9ed7-4bd7-9e42-ab0e89bfac90",
+      "fieldName": "t5_encoder_model"
+    },
+    {
+      "nodeId": "ace0258f-67d7-4eee-a218-6fff27065214",
+      "fieldName": "denoising_start"
+    },
+    {
+      "nodeId": "ace0258f-67d7-4eee-a218-6fff27065214",
+      "fieldName": "num_steps"
```
[the remark this answers is not in the document]
Comment (RyanJDick) [reply invokeai/app/services/workflow_records/default_workflows/FLUX Image to Image.json:36]: Exposed in [627b0bf](https://github.com/invoke-ai/InvokeAI/pull/6798/commits/627b0bf644b01349e03d7ec99a77bb7a77f8118f)

**Pull request merged**